src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java

```diff
@@ -0,0 +1,217 @@
+ */
+public class SQLFunctionsIT extends SQLIntegTestCase {
+
+    @Override
+    protected void init() throws Exception {
+        loadIndex(Index.ACCOUNT);
+    }
+
+    @Test
+    public void functionFieldAliasAndGroupByAlias() throws Exception {
+        String query = "SELECT " +
+                "floor(substring(address,0,3)*20) as key," +
+                "sum(age) cvalue FROM " + TEST_INDEX_ACCOUNT + "/account where address is not null " +
+                "group by key order by cvalue desc limit 10  ";
+        final JSONObject result = executeQuery(query);
+
+
+        IntStream.rangeClosed(0, 9).forEach(i -> {
+                    Assert.assertNotNull(result.query(String.format("/aggregations/key/buckets/%d/key", i)));
+                    Assert.assertNotNull(result.query(String.format("/aggregations/key/buckets/%d/cvalue/value", i)));
+                }
+        );
+    }
+
+    /**
+     * Commented out the following test as alias for normal fields is no longer stored in SearchHit's fields object
+     */
+//    @Test
+//    public void normalFieldAlias() throws Exception {
+//
+//        //here is a bug,csv field with spa
+//        String query = "SELECT " +
+//                "address as key,age from " +
+//                TEST_INDEX_ACCOUNT + "/account where address is not null " +
+//                "limit 10  ";
+//
+//        CSVResult csvResult = getCsvResult(false, query);
+//        List<String> headers = csvResult.getHeaders();
+//        Assert.assertTrue(headers.contains("key"));
+//    }
+
+
+//        CSVResult csvResult = getCsvResult(false, query);
+//        List<String> headers = csvResult.getHeaders();
+//        List<String> contents = csvResult.getLines();
+//        String[] splits = contents.get(0).split(",");
+//        Assert.assertTrue(splits[0].endsWith("--") || splits[1].endsWith("--"));
+
+    @Test
+    public void functionAlias() throws Exception {
+        //here is a bug,if only script fields are included,then all fields will return; fix later
+        String query = "SELECT " +
+                "substring(address,0,3) as key,address from " +
+                TEST_INDEX_ACCOUNT + "/account where address is not null " +
+                "order by address desc limit 10  ";
+
+        assertThat(
+                executeQuery(query),
+                hitAny(both(kvString("/_source/address", equalTo("863 Wythe Place"))).and(kvString("/fields/key/0",
+                        equalTo("863"))))
+        );
+    }
+
+    @Test
+    public void concat_ws_field_and_string() throws Exception {
+        //here is a bug,csv field with spa
+        String query = "SELECT " +
+                " concat_ws('-',age,'-') as age,address from " +
+                TEST_INDEX_ACCOUNT + "/account " +
+                " limit 10  ";
+
+        assertThat(
+                executeQuery(query),
+                hitAny(kvString("/fields/age/0", endsWith("--")))
+        );
+    }
+
+// todo: add the assert phase
```

Comment:
np: uncomment and add `@Ignore` ?

Reply:
Done. Delete in case there is no assert.